source/lib.ts

```diff
@@ -76,18 +76,23 @@ const promisifyStore = (passedStore: LegacyStore | Store): Store => {
 	return new PromisifiedStore()
 }
 
+type OptionsWithLegacyStore = Omit<Partial<Options>, 'store'> & {
+	store?: Store | LegacyStore
+}
```

Comment:
What if we made two different interfaces:

1. `Options`: this is represents the options the user can pass to the rate limiter. The `store` field in the `Options` interface should then be changed to be of type `Store | LegacyStore`. All fields should remain `readonly`, as they currently are.

2. `Configuration`: this represents the parsed configuration that we derive from the options passed. This can exclude deprecated fields like `headers` and `draft_polli_ratelimit_headers`, the `store` field should be of type `Store` and all fields can be read-write.

Reply:
Added the Configuration type. I'm not familiar with what fields are deprecated in this library so I just omitted the ones you mentioned.

Reply:
If we make `Configuration` a separate interface, whose fields do not have the `readonly` modifier, we could get rid of the `@ts-expect-error` directive.

Reply:
@zkldi could you implement this (create a separate interface for `Configuration`) change?

Reply:
Sure, I take it i can just copy over the implementation from Options?

Reply:
No worries, I should have a commit up in the hour.

Reply:
Great, thanks a lot :D

Reply:
It seems a bit redundant to copy the implementation twice. Is there any actual reason Options is marked as readonly?

Reply:
Yuck. Changing this just gives us a different TS error -- ` Unsafe assignment of an any value.  @typescript-eslint/no-unsafe-assignment`. As mentioned before TS is pretty poor when it comes to performing object manipulation.

Changing to a Configuration type doesn't appear to let us remove the ts-expect-error directive.

Reply:
The `Options` interface's properties are marked readonly to ensure we don't change them anyhow - see issue #132

Reply:
The error you describe is a linter error, and you can suppress that separately. I usually try to suppress only lint errors if I have to, not compiler errors.

Reply:
Oh yeah whoops - my bad. I'll submit the commit now then.